File: AdvantageFramework.Core/Database/Views/AlertAttachmentView.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Views
{
    public partial class AlertAttachmentView
    {
        [Column("AttachmentID")]
        public int AttachmentId { get; set; }
        [Column("AlertID")]
        public int AlertId { get; set; }
        [StringLength(100)]
        public string RealFileName { get; set; }
        [Column(TypeName = "smalldatetime")]
        public DateTime AddedDate { get; set; }
        [Required]
        [StringLength(1)]
        public string UserName { get; set; }
        [Required]
        [StringLength(255)]
        public string MimeType { get; set; }
        [Column("EMAILSENT")]
        public bool Emailsent { get; set; }
        [Required]
        [Column("REPOSITORY_FILENAME")]
        [StringLength(200)]
        public string RepositoryFilename { get; set; }
        [Column("DOCUMENT_ID")]
        public int DocumentId { get; set; }
        [Column("FILE_SIZE")]
        public int FileSize { get; set; }
        [Column("DESCRIPTION")]
        [StringLength(200)]
        public string Description { get; set; }
        [Required]
        [Column("USER_CODE")]
        [StringLength(100)]
        public string UserCode { get; set; }
        [Column("PRIVATE_FLAG")]
        public int PrivateFlag { get; set; }
    }
}

File: AdvantageFramework/Database/Views/AlertAttachmentView.vb
Namespace Database.Views

    <Table("WV_ALERTATTACHMENTLIST")>
    Public Class AlertAttachmentView
        Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            AttachmentID
            AlertID
            RealFileName
            AddedDate
            UserName
            MimeType
            EmailSent
            RepositoryFilename
            DocumentID
            FileSize
            Description
            UserCode
            IsPrivate

        End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
        <Required>
        <Column("AttachmentID", Order:=0)>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AttachmentID() As Integer
        <Key>
        <Required>
        <Column("AlertID", Order:=1)>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AlertID() As Integer
        <MaxLength(100)>
        <Column("RealFileName", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property RealFileName() As String
        <Required>
        <Column("AddedDate")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property AddedDate() As Date
        <MaxLength(61)>
        <Column("UserName", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property UserName() As String
        <Required>
        <MaxLength(255)>
        <Column("MimeType", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property MimeType() As String
        <Required>
        <Column("EMAILSENT")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property EmailSent() As Boolean
        <Required>
        <MaxLength(200)>
        <Column("REPOSITORY_FILENAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property RepositoryFilename() As String
        <Required>
        <Column("DOCUMENT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property DocumentID() As Integer
        <Required>
        <Column("FILE_SIZE")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property FileSize() As Integer
        <MaxLength(200)>
        <Column("DESCRIPTION", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property Description() As String
        <Required>
        <MaxLength(100)>
        <Column("USER_CODE", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property UserCode() As String
        <Required>
        <Column("PRIVATE_FLAG")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
        Public Property IsPrivate() As Integer


#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.AttachmentID.ToString

        End Function

#End Region

    End Class

End Namespace
